### PR TITLE
SSR compatibility (through webpack 4)

### LINKED ISSUE #314: SSR: window is not defined
Hi,

Is there anyway to get this to work with SSR?

```
ReferenceError: window is not defined
    at .../src/node_modules/vue-select/dist/vue-select.js:2:756
    at .../src/node_modules/vue-select/dist/vue-select.js:2:688
    at t.exports (.../src/node_modules/vue-select/dist/vue-select.js:2:974)
    at Object.<anonymous> (.../src/node_modules/vue-select/dist/vue-select.js:2:1452)
    at e (.../src/node_modules/vue-select/dist/vue-select.js:1:390)
    at Object.t.exports.t (.../src/node_modules/vue-select/dist/vue-select.js:1:28004)
    at e (.../src/node_modules/vue-select/dist/vue-select.js:1:390)
    at Object.<anonymous> (.../src/node_modules/vue-select/dist/vue-select.js:1:648)
    at e (.../src/node_modules/vue-select/dist/vue-select.js:1:390)
    at t.__esModule.default (.../src/node_modules/vue-select/dist/vue-select.js:1:478)
    at .../src/node_modules/vue-select/dist/vue-select.js:1:483
    at n.(anonymous function).exports (.../src/node_modules/vue-select/dist/vue-select.js:1:143)
    at Object.<anonymous> (.../src/node_modules/vue-select/dist/vue-select.js:1:261)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```

### PR DESCRIPTION
**Pull request by yamsellem**
This pull request bumps webpack from version 1.12 to 4.28 thus allowing to make `vue-select` Server Side Rendering compatible.

Closes #314 and #726
May conflict with #615

A lot of dev dependencies have been updated for compatibility sake, babel, webpack loaders (vue, sass, etc), karma, etc. Build tools too.

What it does:
* build files are split between `vue-select.js` and `vue-select.css` (preventing SSR otherwise)
* build bundle is **~14.5kb** of JS and **~6.5kb** of CSS compared to 33kb in release 2.5.1
* no line of `vue-select` has been updated in the process (only build files)
* all test are passing (the inject mock has been updated — and an ignored test reintroduced) 
* dev environnement is now served through `webpack-dev-server` (and hotreloaded)
* scripts are launched with vanilla webpack
* webpack loaders, configuration and babel have been updated to their latest standards

What it does /not/:
* the linter seems to fail (`eslint` dependency is not provided, introducing a doubt on the matter), so it has not been updated
* tests watcher seems to loop (instead of wait for change), this has not been updated
* the documentation is lightly coupled with the library build, efforts have been made to avoid impacting it, leaving its dependencies as is

In a nutshell, this is a simplification of the build tools, and an update to their up-to-date releases.
I hope this helps — and, if by chance it's released, it will be of great help.

I don't know if the upgrade through Vue cli 3 (#615) is still on the way (it's dated August 2018), but, even so, this PR with vanilla webpack still seems a first step along the way. Adding `vue-select` a useful server side feature — without any code change.

### CODE REVIEW

**Comment by yamsellem** [comment]
The `deploy/netlify`fails on `Command "build:dev" not found`, but this command was missing from `package.json`. Weird.

**Comment by Fusseldieb** [comment]
Thanks @yamsellem . I was trying to upgrade to Webpack 4 and it constantly failed, because the structure from V1 to V2 is heavily different.
Glad that I saw this PR.

I've merged your PR into my repo manually and there are some other fixes too.

Since I am using vue-select actively in my (work) projects, I can't wait until those PR's are accepted, so I just forked and merged it.
